### PR TITLE
Refactor to be compatible with WorkObjects

### PR DESCRIPTION
**Pull request by Djadih**

### CODE REVIEW

**Review by wizeguyy — APPROVED** [review]

File: proxy/blocks.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 
 type BlockTemplate struct {
 	sync.RWMutex
-	Header     *types.Header
+	WorkObject     *types.WorkObject
```
Comment (gameofpointers) [inline proxy/blocks.go:16]: the indentation?

File: proxy/proxy.go [hunk 2]
```diff
@@ -300,17 +250,17 @@ func (s *ProxyServer) fetchBlockTemplate() {
...
 	// Short circuit if the pending header is the same as the current one
-	if t != nil && t.Header != nil && t.Header.SealHash() == pendingHeader.SealHash() {
+	if t != nil && t.WorkObject != nil && t.WorkObject.WorkObjectHeader != nil && t.WorkObject.WorkObjectHeader().SealHash() == pendingWo.SealHash() {
 		return
 	}
 	newTemplate := BlockTemplate{
-		Header: pendingHeader,
-		Target: consensus.DifficultyToTarget(pendingHeader.Difficulty()),
-		Height: pendingHeader.NumberArray(),
+		WorkObject: pendingWo,
```
Comment (gameofpointers) [inline proxy/proxy.go:261]: indentation

File: proxy/blocks.go (hunk 1 above)
Comment (Djadih) [reply proxy/blocks.go:16]: Fixed

File: proxy/proxy.go (hunk 2 above)
Comment (Djadih) [reply proxy/proxy.go:261]: I don't see the issue with this one?

**Pull request merged**